x/lazyledgerapp/types/payformessage.go

```diff
@@ -0,0 +1,333 @@
+			"invalid namespace length: got %d wanted %d",
+			len(msg.GetMessageNameSpaceId()),
+			NamespaceIDSize,
+		)
+	}
+
+	// ensure that the included message is evenly divisble into shares
+	if uint64(len(msg.GetMessage()))%ShareSize != 0 {
+		return fmt.Errorf("Share message must be divisible by %d", ShareSize)
+	}
+
+	// make sure that the message size matches the actual size of the message
+	if msg.MessageSize != uint64(len(msg.Message)) {
+		return fmt.Errorf(
+			"Declared Message size does not match actual Message size, %d vs %d",
+			msg.MessageSize,
+			len(msg.Message),
+		)
+	}
+
+	for _, commit := range msg.MessageShareCommitment {
+		// check that each commit is valid
+		calculatedCommit, err := CreateCommitment(commit.K, msg.GetMessageNameSpaceId(), msg.Message)
+		if err != nil {
+			return err
+		}
+
+		if string(calculatedCommit) != string(commit.ShareCommitment) {
+			return fmt.Errorf("invalid commit for square size %d", commit.K)
+		}
+
+		// check that the signatures are valid
+		bytesToSign, err := msg.GetCommitmentSignBytes(commit.K)
+		if err != nil {
+			return err
+		}
+
+		if !pubK.VerifySignature(bytesToSign, commit.Signature) {
+			return fmt.Errorf("invalid signature for share commitment to square size %d", commit.K)
+		}
+	}
+
+	return nil
+}
+
+// GetSignBytes returns messages bytes that need to be signed in order for the
+// message to be valid
+func (msg *MsgWirePayForMessage) GetSignBytes() []byte {
+	out, err := msg.GetCommitmentSignBytes(SquareSize)
+	if err != nil {
+		// this panic can only be reached if the nmt cannot push bytes onto the
+		// tree while creating the commit. This should never happen, as an error
+		// only occurs when out of order or varying sized namespaces are used,
+		// and we are using an identical namespace when pushing to the nmt
+		// https://github.com/lazyledger/nmt/blob/b22170d6f23796a186c07e87e4ef9856282ffd1a/nmt.go#L250
+		panic(err)
+	}
+	return out
+}
+
+// GetSigners returns the addresses of the message signers
+func (msg *MsgWirePayForMessage) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{sdk.AccAddress(msg.PubKey().Address().Bytes())}
+}
+
+// PubKey returns the public key of the creator of MsgWirePayForMessage
+func (msg *MsgWirePayForMessage) PubKey() *secp256k1.PubKey {
+	return &secp256k1.PubKey{Key: msg.PublicKey}
+}
+
+// GetCommitmentSignBytes generates the bytes that each need to be signed per share commit
+func (msg *MsgWirePayForMessage) GetCommitmentSignBytes(k uint64) ([]byte, error) {
+	sTxMsg, err := msg.SignedTransactionDataPayForMessage(k)
+	if err != nil {
+		return nil, err
+	}
+	return sTxMsg.GetSignBytes(), nil
+}
+
+// SignedTransactionDataPayForMessage use the data in the MsgWirePayForMessage
+// to create a new SignedTransactionDataPayForMessage
+func (msg *MsgWirePayForMessage) SignedTransactionDataPayForMessage(k uint64) (*SignedTransactionDataPayForMessage, error) {
+	// add padding to message if necessary
+	msg.parseMessage()
```

Comment:
The returned value of `parseMessage` is never used. Is that intentional?

Reply:
yes, this kind of relates to the point you made above, the name is confusing. The message needs to be updated after adding padding, so that should be explicit and separate. This has since been changed to simply use a function `PadMessage`.